public/app/features/plugins/sandbox/code_loader.ts

```diff
@@ -58,10 +58,15 @@ export async function getPluginCode(meta: PluginMeta): Promise<string> {
     const response = await fetch('public/' + meta.module + '.js');
     let pluginCode = await response.text();
     pluginCode = patchPluginSourceMap(meta, pluginCode);
+    pluginCode = patchPluginAPIs(pluginCode);
     return pluginCode;
   }
 }
 
+function patchPluginAPIs(pluginCode: string): string {
+  return pluginCode.replace(/window\.location/gi, 'window.locationSandbox');
```

Comment:
This feels a bit scary. Feels like there should be a better way of doing this? Isn't window.location proxied in the read realm?

Reply:
@mckn as noted in description. `window.location` is unforgeable and thus not proxied. This is the only way (and recommended by near membrane too) to achieve this.

Reply:
OK ok, is there any way we could do this without doing the `pluginCode.replace`? Could we expose it as `window.location` instead of `window.locationSandbox`?

Reply:
`window.location` is unforgeable. It means you can't modify it in ANY possible way. We can't proxy it, monkey patch it, alter it, or do anything at all with it.